### ISSUE TITLE
perf: increase default batch size for TinyImageNet GPU examples

### ISSUE THREAD

**Issue by Samyssmile** [Samyssmile/edux#214]
## Problem

The `TinyImagenetCnnAug` example uses **batch size 16** (`BATCH_SIZE = 16` at line 62), which severely underutilizes the GPU. Small batches mean:

1. **High kernel launch overhead** — Each CUDA kernel launch has ~5–15µs overhead. With batch=16 on 64×64 images, the actual compute per kernel is tiny relative to the launch cost. Over 6,250 batches/epoch, this accumulates to significant overhead.

2. **Poor GPU occupancy** — cuBLAS and cuDNN achieve peak throughput with larger matrices. Batch=16 produces small GEMM dimensions (e.g., 16×512 instead of 64×512), leaving many GPU SMs idle.

3. **Suboptimal memory bandwidth utilization** — GPUs are designed for high-throughput streaming. Small batches mean more kernel launches with less data per launch, wasting memory bandwidth on setup rather than compute.

4. **More optimizer steps per epoch** — 6,250 steps at batch=16 vs. 1,563 at batch=64. Each step triggers the moment buffer sync overhead (see #200), gradient clipping, LR scheduling, and callback invocations.

### Memory estimate for TinyImageNet at 64×64

| Batch Size | Input Tensor | Activations (est.) | cuDNN Workspace | Total (est.) |
|---|---|---|---|---|
| 16 | 0.75 MB | ~200 MB | ~50 MB | ~300 MB |
| 32 | 1.5 MB | ~400 MB | ~50 MB | ~500 MB |
| 64 | 3 MB | ~800 MB | ~50 MB | ~900 MB |

Even with 4GB VRAM, batch size 32–64 is feasible for the VGG-style 64×64 model.

### 1. Increase batch size in example

```java
// TinyImagenetCnnAug.java
private static final int BATCH_SIZE = 32;  // was 16
```

Or even 64 if VRAM allows. The example should demonstrate reasonable GPU utilization.

### 2. Adjust learning rate accordingly

When increasing batch size, the learning rate typically needs linear scaling:
- Batch 16 → LR 0.001
- Batch 32 → LR 0.002
- Batch 64 → LR 0.004

The warmup steps should remain proportional to the number of batches.

### 3. Consider making batch size configurable via CLI

```java
int batchSize = Arrays.stream(args)
    .filter(a -> a.startsWith("--batch-size="))
    .mapToInt(a -> Integer.parseInt(a.split("=")[1]))
    .findFirst().orElse(32);
```

This allows users with different VRAM capacities to tune without code changes.

### Expected speedup

**20–30%** from reduced kernel launch overhead and better GPU occupancy. Additionally compounds with other optimizations (fewer optimizer steps means less moment sync overhead per epoch).

### Affected files

- `examples/src/main/java/de/edux/example/tinyimagenet/cuda/TinyImagenetCnnAug.java`

### Acceptance criteria

- [ ] Default batch size increased to 32 (or 64 with OOM fallback documentation)
- [ ] Learning rate and warmup adjusted for new batch size
- [ ] Verified that training converges to comparable accuracy
- [ ] Documented VRAM requirements in class JavaDoc
